tests/regression/lp-1764977/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: Broken snap doesn't break "snap list" and other operations
+details: |
+    Snap can enter the "broken" state when its meta/snap.yaml cannot be found.
+    Typically this means the snap is unmounted or that "snap try"-mode snap was
+    moved on disk and the symlink in /var/lib/snapd/snaps/ is no longer
+    working.
+
+    Broken snaps should not affect snapd operations though, as that forms a
+    denial-of-service attack and may keep machines from refreshing core.
+execute: |
+    cp -a  $TESTSLIB/snaps/test-snapd-sh .
+    snap try test-snapd-sh
+
+    snap list | MATCH test-snapd-sh
+    mv test-snapd-sh test-snapd-sh.moved
+    snap list # should not break
+    snap list test-snapd-sh | MATCH broken
+
+    mv test-snapd-sh.moved test-snapd-sh
+    snap list # should still just work
+    snap list test-snapd-sh | MATCH -v broken
```

Comment:
Probably makes sense to update the test with a case where `meta/snap.yaml` contents are garbage

Reply:
I think that should be done in a big collective test. This is specifically regression testing the thing I ran into.